carl_interactive_manipulation/src/create_parking_spots.cpp

```diff
@@ -0,0 +1,128 @@
+#include <carl_interactive_manipulation/create_parking_spots.hpp>
+
+/** creates clickable navigation goals infront of furniture*/
+int main(int argc, char** argv){
+  ros::init(argc,argv,"create_parking_spots");
+  ros::NodeHandle node;
+  ros::Rate rate(10.0);
+
+  ParkingSpots parkingSpots;
+  parkingSpots.createParkingSpots();
+  
+  return 0;
+}
+
+ParkingSpots::ParkingSpots() : client_("move_base",true),server_("parking_markers") {
+}
+
+void ParkingSpots::createParkingSpots(){
+  ROS_INFO("waiting for server...");
+  
+  bool status = client_.waitForServer();
+  if (!status){
+    ROS_INFO("Couldn't connect in time...");
+  }
+  else {
+
+  ROS_INFO("connected to server");
+
+  urdf::Model ilab;
+
+  //load the urdf with all the furniture off the param server
+  if (!ilab.initParam("/ilab_description")){
+    ROS_INFO("couldn't find /ilab_description on param server.");
+  }
+  else {
+
+    std::map<std::string, boost::shared_ptr<urdf::Link> > links = ilab.links_;
+    std::map<std::string, boost::shared_ptr<urdf::Link> >::iterator itr;
+    
+    //go through all links and filter out the ones that end in "nav_goal_link"
+    for(itr = links.begin(); itr != links.end(); itr++) {
+      std::string link_name = itr->first;
+      if (isNavGoal(link_name)){
+        visualization_msgs::InteractiveMarker marker = createParkingSpot(link_name);
+        ROS_INFO("creating marker %s",link_name.c_str());
+        server_.insert(marker,boost::bind(&ParkingSpots::onClick, this, _1));
+      }
+    }
+
+    ROS_INFO("creating clickable nav goals...");
+
+    //when these are called the markers will actually appear
+    server_.applyChanges();  
+    ros::spin();
```

Comment:
Yeah it should work if you put the spin at the end of your main.  Also your main function declares a NodeHandle and a Rate that you're not using, so you could take that out.  And finally, instead of having an empty constructor followed by immediately calling createParkingSpots, why not instead just put the createParkingSpots code in the constructor?  Either that, or have the constructor call createParkingSpots and take the extra call out of your main function.